Compaction throws NullPointerException but returns 202 Accepted

In the context of cass-operator, it seems that invoking the `/api/v1/ops/tables/compact` endpoint returns a 202 Accepted, but cass-operator logs show that the task failed due to a NullPointerException:

`"error": "task failed: java.lang.NullPointerException"`

```
ERROR [epollEventLoopGroup-5-3] 2024-02-06 12:44:25,846 NodeOpsProvider.java:97 - Task 8a0b3e1b-fd2a-4514-ae76-07ccd8214a7e execution failed
java.util.concurrent.CompletionException: java.lang.NullPointerException
        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314)
        at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319)
        at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1739)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.NullPointerException: null
        at com.datastax.mgmtapi.NodeOpsProvider.lambda$forceKeyspaceCompaction$6(NodeOpsProvider.java:396)
        at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1736)
        ... 3 common frames omitted
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,9 @@ Changelog for Management API, new PRs should update the `main / unreleased` sect
 
 ## unreleased
 
+* [BUGFIX] [#448](https://github.com/k8ssandra/management-api-for-apache-cassandra/issues/448) Scrub did not work when targetting all the keyspaces
+* [BUGFIX] [#437](https://github.com/k8ssandra/management-api-for-apache-cassandra/issues/437) Compaction threw NPE if used without target TokenRange
+
 ## v0.1.72 (2023-12-19)
 
 * [BUGFIX]  [#434](https://github.com/k8ssandra/management-api-for-apache-cassandra/issues/434) Temporarily disable arm64 support for DSE images

--- a/management-api-agent-common/src/main/java/com/datastax/mgmtapi/NodeOpsProvider.java
+++ b/management-api-agent-common/src/main/java/com/datastax/mgmtapi/NodeOpsProvider.java
@@ -94,6 +94,7 @@ public class NodeOpsProvider {
     resultMap.put("submit_time", String.valueOf(jobWithId.getSubmitTime()));
     resultMap.put("end_time", String.valueOf(jobWithId.getFinishedTime()));
     if (jobWithId.getStatus() == Job.JobStatus.ERROR) {
+      logger.error("Task " + jobId + " execution failed", jobWithId.getError());
       resultMap.put("error", jobWithId.getError().getLocalizedMessage());
     }
 

--- a/management-api-server/src/main/java/com/datastax/mgmtapi/resources/TableOpsResources.java
+++ b/management-api-server/src/main/java/com/datastax/mgmtapi/resources/TableOpsResources.java
@@ -217,7 +217,7 @@ public class TableOpsResources extends BaseResources {
                 "CALL NodeOps.forceKeyspaceCompaction(?, ?, ?, ?)",
                 compactRequest.splitOutput,
                 keyspaceName,
-                compactRequest.tables,
+                tables,
                 false);
           }
 

--- a/management-api-server/src/main/java/com/datastax/mgmtapi/resources/v1/TableOpsResources.java
+++ b/management-api-server/src/main/java/com/datastax/mgmtapi/resources/v1/TableOpsResources.java
@@ -237,7 +237,7 @@ public class TableOpsResources extends BaseResources {
                       "CALL NodeOps.forceKeyspaceCompaction(?, ?, ?, ?)",
                       compactRequest.splitOutput,
                       keyspaceName,
-                      compactRequest.tables,
+                      tables,
                       true))
               .build();
         });

